core/common/src/main/java/alluxio/util/UnderFileSystemUtils.java

```diff
@@ -67,5 +67,36 @@ public static void touch(final String path) throws IOException {
     os.close();
   }
 
+  /**
+   * Strips the suffix if it exists. This method will leave keys without a suffix unaltered.
+   *
+   * @param key the key to strip the suffix from
+   * @param suffix suffix to remove
+   * @return the key with the suffix removed, or the key unaltered if the suffix is not present
+   */
+  public static String stripSuffixIfPresent(final String key, final String suffix) {
```

Comment:
Is there a unit test for this method?

Also, is there a reason this is called `key` and `stropPrefixIfPresent` is `path`? Being consistent would be nice if possible.